rpaas/consul_manager.py

```diff
@@ -152,18 +155,25 @@ def remove_lua(self, instance_name, lua_module_name, lua_module_type):
 
     def add_server_upstream(self, instance_name, upstream_name, server):
         servers = self.list_upstream(instance_name, upstream_name)
+        if not server:
+            return
         if isinstance(server, list):
+            for idx, _ in enumerate(server):
+                server[idx] = ":".join(map(str, filter(None, self._host_from_destination(server[idx]))))
```

Comment:
What's `filter(None, ...` trying to achieve here? Isn't this the same as simply using the tuple returned by `_host_from_destination`?

Reply:
In some situations destinations doesn't contain port.

Reply:
ah, okay